gems/smithy/lib/smithy/rivet.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+module Smithy
+  # Base class that all rivets must inherit from. Includes hooks for the forging process.
+  class Rivet
+    include Thor::Base
+    include Thor::Actions
+
+    # @param [Plan] plan The plan that is being executed.
+    def initialize(plan)
+      @plan = plan
+      # Necessary for Thor::Base and Thor::Actions
+      self.options = { force: true }
+      self.destination_root = plan.options[:destination_root]
+    end
+
+    # Called to determine if the rivet should be applied for this model.
+    # @param [Hash] service Service shape
+    # @return [Boolean] (true) True if the rivet should be applied, false otherwise.
+    def for?(service) # rubocop:disable Lint/UnusedMethodArgument
```

Comment:
Is this just the Service shape? If so, I'm not sure its sufficient?  

EDIT: Nvm - I guess you have access to the @plan and therefore the model as well right?

Reply:
It's just the service shape. We can improve this if we need more scope. I had removed `plan` from the instance variable. Why else would you need the full model?